core/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -195,16 +220,15 @@ private boolean setResponseFailureIfIndexMatches(AtomicArray<BulkItemResponse> r
      * @see #doExecute(BulkRequest, org.elasticsearch.action.ActionListener)
      */
     public void executeBulk(final BulkRequest bulkRequest, final ActionListener<BulkResponse> listener) {
-        final long startTime = System.currentTimeMillis();
-        executeBulk(bulkRequest, startTime, listener, new AtomicArray<>(bulkRequest.requests.size()));
+        final long startTimeNanos = relativeTime();
```

Comment:
You assume here that the time provider returns time in nanoseconds but given the scope of the time provider I think it is fine.